tests/director/tests/formation_mapping_api_test.go

```diff
@@ -489,31 +489,14 @@ func Test_UpdateStatus(baseT *testing.T) {
 		formationAssignmentID := getFormationAssignmentIDByTargetTypeAndSourceID(t, assignmentsPage, graphql.FormationAssignmentTypeRuntime, app.ID)
 		t.Logf("successfully listed FAs for formation ID: %q", formationID)
 
-		certSecuredHTTPClientWithDifferentTenant := getHTTPCertClientWithCustomSubject(t, ctx, conf, conf.ExternalCertTestIntSystemOUSubaccount, conf.ExternalCertTestIntSystemCommonName)
+		// Prepare provider external client certificate and secret and Build graphql director client configured with certificate
+		providerClientKey, providerRawCertChain := certprovider.NewExternalCertFromConfig(t, ctx, conf.ExternalCertProviderConfig, true)
+		certSecuredHTTPClientWithDifferentTenant := gql.NewCertAuthorizedHTTPClient(providerClientKey, providerRawCertChain, conf.SkipSSLValidation)
```

Comment:
Just of curiosity, did the test work before without the new changes? Because with the new changes, the default "provider" account is different but most probably it should work without modifying the test here?

Or you wanted to remove the integration system "subject mapping"? Locally most probably there is no difference but in productive setup we won't used the mapping anymore rather than before where "subject mapping" was used both locally and productively?

Reply:
The test was failing with the new changes because it was issuing a cert that is part of the subject mapping. Just not the Super Admin, but the Integration System one in the default mapping. Therefore with the new change no tenant was populated (as there is no tenant header) and therefore the call failed with Tenant Required error. With this change we issue a cert that is not part of the subject mapping and is also not owner of the FAs and therefore the test works.